crates/bevy_reflect/src/null.rs

```diff
@@ -0,0 +1,59 @@
+use alloc::boxed::Box;
+use core::fmt::Debug;
+
+use crate::{
+    ApplyError, PartialReflect, Reflect, ReflectMut, ReflectOwned, ReflectRef, TypeInfo, TypePath,
+};
+
+#[derive(Clone, Copy, Debug, Default, TypePath)]
+pub struct Null;
```

Comment:
This very much needs docs explaining why it exists.

Reply:
> This very much needs docs explaining why it exists.

I think I may have gotten a bit ahead of myself by writing code before fleshing out what should be done and why.

I've started a discussion. https://github.com/bevyengine/bevy/discussions/18713